### DISCUSSION TITLE
ARM82 的编译器开关会对结果的精度产生影响，建议可选开启

### DISCUSSION THREAD

**Discussion by niinorz** [Tencent/ncnn#3933] [General]
由于我需要在多个平台间切换，有一些平台不支持ARM v8.2，所以我尝试将 NCNN_COMPILER_SUPPORT_ARM82_FP16 和 NCNN_COMPILER_SUPPORT_ARM82_FP16_DOTPROD 设置为 FALSE。然后我比较了打开和关闭之后的结果，发现结果的精度差异多的能到 0.02
现在版本是只要编译器支持就自动开启，是否可以增加一个大的开关来控制是否启用
[diff.txt](https://github.com/Tencent/ncnn/files/8939710/diff.txt)

补充说明一下：我们的精度差异需要控制在 1/220 以内，也就是 < 0.0045

**Comment by nihui** [comment]
```cpp
opt.use_fp16_storage = false;
opt.use_fp16_arithmetic = false;
```

load_param 前设置开关，可以关闭自动的 fp16 存储和运算，退回到 fp32 模式

**Reply by niinorz** [reply]
嗯...现在opt选项开和关，加上编译选项关掉，有三种输出结果，关掉opt选项的输出精度差异会更大一些

**Comment by niinorz** [comment]
前面的误差数据是使用使用 3a0ba55c6f6ed35634f9af44ec06e6f07a46a8d3 版本，如果使用 20220420 的版本，误差最大会到 0.2

**Comment by niinorz** [comment]
遇到灵异事件了，今天基于 20220216 版本，想再测一下 编译选项 和 option 开关不同情况的对比
结果，除了（编译选项为 TRUE，且option 为 true）的情况，计算结果为 nan
其他三种组合的情况下，结果都是一样的
待我研究一下再来
